bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseAudioAudioSink.java

```diff
@@ -74,69 +78,123 @@ public void process(@Nullable AudioStream audioStream)
-                            long millisSecondTimedToSendAudioData = Duration.between(start, end).toMillis();
-                            if (millisSecondTimedToSendAudioData < normalizedPCMStream.getDuration()) {
-                                CompletableFuture<@Nullable Void> soundPlayed = new CompletableFuture<>();
-                                long timeToWait = normalizedPCMStream.getDuration() - millisSecondTimedToSendAudioData;
-                                logger.debug("Some time to let the system play sound : {}", timeToWait);
-                                scheduler.schedule(() -> soundPlayed.complete(null), timeToWait, TimeUnit.MILLISECONDS);
-                                return soundPlayed;
-                            } else {
-                                return CompletableFuture.completedFuture(null);
-                            }
-                        } else {
-                            // We have a second method available to guess the duration, and it is during transfer
-                            Long timeStampEnd = audioSinkUtils.transferAndAnalyzeLength(normalizedPCMStream,
-                                    clientSocketLocal.getOutputStream(), TARGET_FORMAT);
-                            CompletableFuture<@Nullable Void> soundPlayed = new CompletableFuture<>();
-                            if (timeStampEnd != null) {
-                                long now = System.nanoTime();
-                                long timeToWait = timeStampEnd - now;
-                                if (timeToWait > 0) {
-                                    scheduler.schedule(() -> soundPlayed.complete(null), timeToWait,
-                                            TimeUnit.NANOSECONDS);
-                                }
-                                return soundPlayed;
-                            } else {
-                                return CompletableFuture.completedFuture(null);
+                    AudioWaveUtils.removeFMT(pcmSignedStream);
+                } catch (IOException e) {
+                    logger.warn("IOException trying to remove wav header: {}", e.getMessage());
+                }
+            }
+            if (pcmSignedStream instanceof SizeableAudioStream sizeableAudioStream) {
+                long length = sizeableAudioStream.length();
+                var audioFormat = pcmSignedStream.getFormat();
+                long byteRate = (Objects.requireNonNull(audioFormat.getBitDepth()) / 8)
+                        * Objects.requireNonNull(audioFormat.getFrequency())
+                        * Objects.requireNonNull(audioFormat.getChannels());
+                float durationInSeconds = (float) length / byteRate;
+                duration = Math.round(durationInSeconds * 1000);
+                logger.debug("Duration of input stream : {}", duration);
+            }
+        } else {
+            try {
+                var convertedInputStream = new ConvertedInputStream(audioStream);
```

Comment:
Inside that try block the only line that can throw an exception is that one, so there is nothing to release in the catch block if the exception is thrown. Then it's assigned to variable `pcmSignedStream` and after the last changes it should be correctly closed after usage or in case of errors. I think it's consistent now.